File: src/Entity/Type.php
<?php

namespace App\Entity;

use App\Repository\TypeRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: TypeRepository::class)]
class Type
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id;

    #[ORM\Column(type: 'string', length: 255)]
    private ?string $mode = null;
    #[ORM\OneToMany(targetEntity: Test::class, mappedBy: "type")]
    private $tests;
    public function __construct()
    {
        $this->tests = new ArrayCollection();
    }

    public function getTests(): Collection
    {
        return $this->tests;
    }

    public const MODES = ['Présentiel' => 'présentiel', 'En ligne' => 'en_ligne'];

    public function getId(): ?int
    {
       return $this->id;
    }

    public function getMode(): ?string
    {
        return $this->mode;
    }

    public function setMode(string $mode): self
    {
        $this->mode = $mode;

        return $this;
    }

}

File: jdbca23/src/main/java/esprit/tn/entities/Type.java
package esprit.tn.entities;

public class Type {
    private int id;
    private String mode;

    public Type() {
    }

    public Type(int id, String mode) {
        this.id = id;
        this.mode = mode;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getMode() {
        return mode;
    }

    public void setMode(String mode) {
        this.mode = mode;
    }

    @Override
    public String toString() {
        return "Type{" +
                "id=" + id +
                ", mode='" + mode + '\'' +
                '}';
    }
} 